src/rax.c

```diff
@@ -731,7 +731,7 @@ int raxGenericInsert(rax *rax, unsigned char *s, size_t len, void *data, void **
 
         if (j == 0) {
             /* 3a: Replace the old node with the split node. */
-            if (h->iskey) {
+            if (h->iskey && !h->isnull) {
```

Comment:
This kind of changes is hard to verify when reviewing.

If `h->isnull` is always 0 here, better write an assert and a code comment explaining why it's always zero.

Reply:
The correct reason is not that h->isnull is always zero.
You can see that the semantics of the judgment here is that when raxNode h is key, assign data to splitnode. Obviously, it doesn't make sense if h->isnull==true. You can see that there are many such judgments in rax.c, such as line746, 796

Reply:
Ah, I understand.

If `h->isnull == 1`, then `raxGetData(h)` returns NULL, and then `raxSetData(splitnode,ndata)` sets `splitnode->isnull = 1`.

But where is `splitnode->isnull = 1` set after this change?

Reply:
I'm sorry, It's my fault.